src/common/include/gudhi/CGAL_Random_Singleton.h

```diff
@@ -0,0 +1,51 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Vincent Rouvreau
+ *
+ *    Copyright (C) 2025 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef CGAL_RANDOM_SINGLETON_H_
+#define CGAL_RANDOM_SINGLETON_H_
+
+#include <CGAL/Random.h>
+#include <CGAL/version.h>  // for CGAL_VERSION_NR
+
+// Make compilation fail - required for external projects - https://github.com/GUDHI/gudhi-devel/issues/10
+#if CGAL_VERSION_NR < 1041101000
+# error random_point_generators is only available for CGAL >= 4.11
+#endif
+
+namespace Gudhi {
+
+  class CGAL_Random_Singleton {
+   public:
+    static CGAL::Random* get() {
+      if(singleton_ == nullptr){
+        singleton_ = new CGAL::Random();
+      }
+      return singleton_;
+    }
+    static CGAL::Random* get(unsigned int seed) {
+      set_seed(seed);
+      return singleton_;
+    }
+    static void set_seed(unsigned int seed) {
+      if(singleton_ != nullptr){
+        delete singleton_;
+      }
+      singleton_ = new CGAL::Random(seed);
+    }
+   private:
+    CGAL_Random_Singleton() {};
+    static CGAL::Random* singleton_;
```

Comment:
If we are going to use CGAL for random number generation, why create a new singleton instead of using the one in CGAL? (possibly with wrapper functions in gudhi if we want to be able to change the backend at some point, but still using the object returned by `CGAL::get_default_random()`)
Or if we roll our own, it would make more sense to base in on the standard (since C++11) `<random>` instead of CGAL.